connectors/sources/salesforce.py

```diff
@@ -773,6 +1076,104 @@ def _format_address(self, address):
         ]
         return ", ".join([a for a in address_fields if a])
 
+    def _format_time_for_sort(self, time_body_pair):
+        if time_body_pair[0] is None:
+            return ""
+
+        return time_body_pair[0]
+
+    def _format_case_body(self, case):
+        time_body_pairs = []
+        time_body_pairs.append([case.get("CreatedDate"), case.get("Description")])
+
+        case_comments = case.get("CaseComments", {}) or {}
+        for comment in case_comments.get("records", []):
+            time_body_pairs.append(
+                [comment.get("CreatedDate"), comment.get("CommentBody")]
+            )
+
+        email_messages = case.get("EmailMessages", {}) or {}
+        for email in email_messages.get("records", []):
+            subject = email.get("Subject", "") or ""
+            text_body = email.get("TextBody", "") or ""
+            time_body_pairs.append(
+                [email.get("MessageDate"), f"{subject}\n{text_body}"]
+            )
+
+        case_feeds = case.get("Feeds", []) or []
+        for feed in case_feeds:
+            title = feed.get("Title", "") or ""
+            body = feed.get("Body", "") or ""
+            time_body_pairs.append([feed.get("CreatedDate"), f"{title}\n{body}"])
+
+            feed_comments = feed.get("FeedComments", {}) or {}
+            for comment in feed_comments.get("records", []):
+                time_body_pairs.append(
+                    [comment.get("CreatedDate"), comment.get("CommentBody")]
+                )
+
+        time_body_pairs = sorted(time_body_pairs, key=self._format_time_for_sort)
+        return "\n\n".join(
+            filter(
+                None, [str(x[-1]).strip() for x in time_body_pairs if x[-1] is not None]
+            )
+        )
```

Comment:
What's going on here? Might be worth to make a function or add a comment explaining this

Reply:
Yeah this was much prettier in the original ruby code. I'll make it easier to understand.

Reply:
@artem-shelkovnikov I broke it up and added comments